mlflow/tracing/export/mlflow.py

```diff
@@ -91,11 +91,14 @@ def _serialize_inputs_outputs(self, input_or_output: Optional[Dict[str, Any]]) -
         if not input_or_output:
             return ""
 
-        try:
-            serialized = json.dumps(input_or_output)
-        except TypeError:
-            # If not JSON-serializable, use string representation
-            serialized = str(input_or_output)
+        if type(input_or_output) is str:
+            serialized = input_or_output
+        else:
+            try:
+                serialized = json.dumps(input_or_output)
+            except TypeError:
+                # If not JSON-serializable, use string representation
+                serialized = str(input_or_output)
```

Comment:
```suggestion
            serialized = json.dumps(input_or_output, default=str)
```

Can we use the `default` argument?

Reply:
I think we should use default=str in try block, and still keep the catch to use str(). If it's a weird object then I guess default=str doesn't solve the problem

Reply:
makes sense, will add `default=str` in try-catch!

Reply:
hmm actually if `default=str` doesn't work even `str(input_or_output)` doesn't work.... I think we should just log and return empty string in except.